addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1148,6 +1149,349 @@ registry.anchorSlide = publicWidget.Widget.extend({
     },
 });
 
+
+registry.countdown = publicWidget.Widget.extend({
+    selector: '.s_countdown',
+    xmlDependencies: ['/website/static/src/xml/website.s_countdown.xml'],
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.hereBeforeTimerEnds = false;
+        this.endAction = this.$el[0].dataset.endAction;
+        this.endTime = parseInt(this.$el[0].dataset.endTime);
+        this.size = this.$el[0].dataset.size;
+        this.display = this.$el[0].dataset.display;
+
+        this.layout = this.$el[0].dataset.layout;
+        this.shape = this.$el[0].dataset.shape;
+        this.shapeBackground = this.$el[0].dataset.shapeBackground;
+        this.progressBarStyle = this.$el[0].dataset.progressBarStyle;
+        this.progressBarWeight = this.$el[0].dataset.progressBarWeight;
+
+        this.diff = {
+            'days': {
+                'canvas': this.$('.s_countdown_days')[0],
+                'total': 15, // TODO RATIO DAYS ? Based on how many days ?
+                'visible': this._isUnitVisible('d'),
+                'label': _t('Days'),
+            },
+            'hours': {
+                'canvas': this.$('.s_countdown_hours')[0],
+                'total': 24,
+                'visible': this._isUnitVisible('h'),
+                'label': _t('Hours'),
+            },
+            'minutes': {
+                'canvas': this.$('.s_countdown_minutes')[0],
+                'total': 60,
+                'visible': this._isUnitVisible('m'),
+                'label': _t('Minutes'),
+            },
+            'seconds': {
+                'canvas': this.$('.s_countdown_seconds')[0],
+                'total': 60,
+                'visible': this._isUnitVisible('s'),
+                'label': _t('Seconds'),
+            },
+        };
+
+        this._render();
+
+        // Needed if edit mode entered after countdown end
+        this.$('.s_countdown_end_redirect_message').remove();
```

Comment:
No, it should probably remove in the `destroy` function, so that it is indeed removed when edit mode entered and saved.

Reply:
I know, it was there in the first place.
But since `_render()` is called directly once before being called in the `setInterval` (so it does not wait 1 second after start to be rendered), it would render the redirect end message twice, one directly and one in the first `setInterval` occurence.
I conveniently inserted that line between the first `_render()` and the `setInterval` to avoid that issue.

But I just changed it as you requested, now it is on the destroy method, but needed to condition the rendering in the `_handleEndCountdown()` (called by `_render()`) to not add it again if already added:
```javascript
if (!this.$('.s_countdown_end_redirect_message').length)
```